index.js

```diff
@@ -185,49 +185,52 @@ function parseToken (buffer) {
   return result
 }
 
-const numMap = {
-  1: 'If-Match',
-  3: 'Uri-Host',
-  4: 'ETag',
-  5: 'If-None-Match',
-  6: 'Observe',
-  7: 'Uri-Port',
-  8: 'Location-Path',
-  9: 'OSCORE',
-  11: 'Uri-Path',
-  12: 'Content-Format',
-  14: 'Max-Age',
-  15: 'Uri-Query',
-  16: 'Hop-Limit',
-  17: 'Accept',
-  19: 'Q-Block1',
-  20: 'Location-Query',
-  23: 'Block2',
-  27: 'Block1',
-  28: 'Size2',
-  31: 'Q-Block2',
-  35: 'Proxy-Uri',
-  39: 'Proxy-Scheme',
-  60: 'Size1',
-  258: 'No-Response',
-  2049: 'OCF-Accept-Content-Format-Version',
-  2053: 'OCF-Content-Format-Version'
+const OPTIONS_BY_NAME = {
+  "If-Match": 1,
+  "Uri-Host": 3,
+  "ETag": 4,
+  "If-None-Match": 5,
+  "Observe": 6,
+  "Uri-Port": 7,
+  "Location-Path": 8,
+  "OSCORE": 9,
+  "Uri-Path": 11,
+  "Content-Format": 12,
+  "Max-Age": 14,
+  "Uri-Query": 15,
+  "Hop-Limit": 16,
+  "Accept": 17,
+  "Q-Block1": 19,
+  "Location-Query": 20,
+  "Block2": 23,
+  "Block1": 27,
+  "Size2": 28,
+  "Q-Block2": 31,
+  "Proxy-Uri": 35,
+  "Proxy-Scheme": 39,
+  "Size1": 60,
+  "No-Response": 258,
+  "OCF-Accept-Content-Format-Version": 2049,
+  "OCF-Content-Format-Version": 2053
 }
 
-const optionNumberToString = (function genOptionParser () {
-  let code = Object.keys(numMap).reduce(function (acc, key) {
-    acc += 'case ' + key + ':\n'
-    acc += '  return \'' + numMap[key] + '\'\n'
-
-    return acc
-  }, 'switch(number) {\n')
+const OPTIONS_BY_NUMBER = new Array(2054)
+for (const key in OPTIONS_BY_NAME) {
+  const number = OPTIONS_BY_NAME[key]
+  OPTIONS_BY_NUMBER[number] = key
+}
 
-  code += 'default:\n'
-  code += 'return \'\' + number'
-  code += '}\n'
+function optionNumberToString (number) {
+  const string = OPTIONS_BY_NUMBER[number]
+  if (string) return string
```

Comment:
If `0` is intended to be supported, you could use `string !== undefined`.

Little to no difference to the benchmark.
```js
optionNumberToStringNew: 1.537s
```

Reply:
Thank you for your contribution, @dcousens! :) This is a great improvement over the old version. As to this particular check: As `0` is a reserved number, I think it is safe to keep it as is :)

Reply:
Sorry,  this comment should have been on `optionNumberToString`, but the implications of comment remain the same.